src/Pages/EmailPreferences/EmailPreferences.js

```diff
@@ -0,0 +1,152 @@
+      const userEmailParam = urlParams.get('user');
+      setUserEmail(userEmailParam);
+      if (!userEmailParam) {
+        window.location = window.location.origin;
+      } else {
+        const response = await getUserData(urlParams.get('user'));
+        if (!response.error) {
+          setUserFirstName(response.responseData.firstName);
+          setUserLastName(response.responseData.lastName);
+          setIsOptedIntoEmails(response.responseData.emailOptIn);
+        } else {
+          setErrorLoading(true);
+        }
+        setLoading(false);
+      }
+    };
+
+    fetchData();
+  }, []);
+
+  const handleSave = async () => {
+    const response = await setUserEmailPreference(
+      userEmail,
+      isOptedIntoEmails
+    );
+    if (!response.error) {
+      setMessage('Yay! You changed your email preference!');
+      setButtonColor(GREEN);
+      setTimeout(() => {
+        setButtonColor(BLUE);
+        setMessage('Submit');
+      }, 3000);
+    } else {
+      setButtonColor(RED);
+      setMessage('Unable to submit your preference, try reloading? :(');
+    }
+  };
+
+  if (errorLoading) {
+    return (
+      <div className='email-preferences'>
+        {/*
+          dumb hacks because the navbar otherwise covers up the below content
+        */}
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
+        <br />
+        <Container
+          style={{
+            textAlign: 'center',
+          }}
+        >
+          <h3>
+            The email page was unable to load at this time.
+          </h3>
+          <h4>
+
+            If the problem persists after reloading this page, please contact us on Discord
+            (<a href='https://discord.gg/KZCKCEz5YA'>invite link</a>)
+          </h4>
+        </Container>
+      </div>
+    );
+  }
+
+  return (
+    <div className='email-preferences'>
+      <Header title='Email Preferences' />
+      <Container
+        style={{
+          textAlign: 'center',
+        }}
+      >
+        {!loading && <>
+          <h1>
+            Hello {userFirstName} {userLastName},
+          </h1>
+          <p>
+            What would you like to hear about?
+          </p>
+          <Col>
+            <Row>
+              <Label check>
+                <Input
+                  type='radio'
+                  name='radio1'
+                  checked={isOptedIntoEmails}
+                  onChange={() => setIsOptedIntoEmails(true)}
+                />
+                {' '}I would like to continue to recieve club update emails from SCE
+              </Label>
+            </Row>
+            <Row>
+              <Label check>
+                <Input
+                  type='radio'
+                  name='radio1'
+                  checked={!isOptedIntoEmails}
+                  onChange={() => setIsOptedIntoEmails(false)}
+                />
+                {' '}No thanks, I would like to unsubscribe from all emails (best choice)
```

Comment:
is this really the best choice :eyes:

Reply:
yes
![image](https://github.com/SCE-Development/Clark/assets/36345325/90a8e615-90a3-42a0-bea0-54b538ffc30b)